aws/resource_aws_appsync_resolver_test.go

```diff
@@ -0,0 +1,559 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsAppsyncResolver_ResponseTemplate(t *testing.T) {
+	rName := fmt.Sprintf("tfacctest%d", acctest.RandInt())
+	resourceName := "aws_appsync_resolver.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAppsyncResolverDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppsyncResolver_ResponseTemplate(rName, 200),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncResolverExists(resourceName),
+					resource.TestMatchResourceAttr(resourceName, "response_template", regexp.MustCompile(`ctx\.result\.statusCode == 200`)),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAppsyncResolver_ResponseTemplate(rName, 201),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncResolverExists(resourceName),
+					resource.TestMatchResourceAttr(resourceName, "response_template", regexp.MustCompile(`ctx\.result\.statusCode == 201`)),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsAppsyncResolver_multipleResolvers(t *testing.T) {
+	rName := fmt.Sprintf("tfacctest%d", acctest.RandInt())
+	resourceName := "aws_appsync_resolver.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsAppsyncResolverDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAppsyncResolver_multipleResolvers(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsAppsyncResolverExists(resourceName+"1"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"2"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"3"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"4"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"5"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"6"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"7"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"8"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"9"),
+					testAccCheckAwsAppsyncResolverExists(resourceName+"10"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsAppsyncResolverDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).appsyncconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_appsync_resolver" {
+			continue
+		}
+
+		apiID, typeName, fieldName, err := decodeAppsyncResolverID(rs.Primary.ID)
+
+		if err != nil {
+			return err
+		}
+
+		input := &appsync.GetResolverInput{
+			ApiId:     aws.String(apiID),
+			TypeName:  aws.String(typeName),
+			FieldName: aws.String(fieldName),
+		}
+
+		_, err = conn.GetResolver(input)
+		if err != nil {
+			if isAWSErr(err, appsync.ErrCodeNotFoundException, "") {
+				return nil
```

Comment:
For test configurations that include multiple `aws_appsync_resolver` resources, this should `continue` the `for` loop instead of exiting immediately to ensure that all of the resources have been properly deleted.

```suggestion
				continue
```